File: pset6/readability/readability.py
from cs50 import get_string

letters = 0
words = 0
sentences = 0
string_text = get_string("Text: ")
print(string_text.lower())

for i in range(len(string_text)):
    char = string_text[i]
    if (char.isalpha()):
        letters += 1
    elif (char == " "):
        words += 1
    elif (char == '.' or char == '?' or char == '!'):
        sentences += 1

words += 1
L = letters / words * 100
S = sentences / words * 100
index = round(0.0588 * L - 0.296 * S - 15.8)
print(letters, words, sentences, index)
if (index < 16 and index >= 0):
    print(f"Grade {index}")
elif (index >= 16):
    print(f"Grade 16+")
else:
    print("Before Grade 1")

File: pset2/readability/readability.c
#include<cs50.h>
#include<stdio.h>
#include<string.h>
#include<math.h>

int main(void)
{
    string text = get_string("Text: ");
    int letters = 0;
    int words = 0;
    int sentences = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] >= 'A' || text[i] >= 'a')
        {
            letters++;
        }
        else if (text[i] == ' ')
        {
            words++;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    words++;
    float L = (float) letters / (float) words * 100;
    float S = (float) sentences / (float) words * 100;
    float index_raw = 0.0588 * L - 0.296 * S - 15.8;
    int index = round(index_raw);
    //printf("%d\n", index);

    if (index < 16 && index >= 0)
    {
        printf("Grade %d\n", index);
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Before Grade 1\n");
    }

    printf("%d, %d, %d, %d", letters, words, sentences, index);

}